async-nats/src/lib.rs

```diff
@@ -809,15 +809,7 @@ impl Client {
         self.publish_with_reply(subject, inbox, payload).await?;
         self.flush().await?;
         match sub.next().await {
-            Some(message) => {
```

Comment:
I disagree, IMHO `no responders` is an error in request/reply pattern in  Core NATS.

Reply:
We, again, miss proper error handling. Would be easy to check for no responders in `context.request()` then...

Now I guess we should just check for (oh so ugly) string in error and map it to `StatusCode::NO_RESPONDERS` ?

Reply:
> Now I guess we should just check for (oh so ugly) string in error and map it to StatusCode::NO_RESPONDERS ?

Adding status codes to `jetstream::Error`:

```rust
impl Error {
  pub fn status() -> Option<StatusCode> {
    // ...
  }

  pub fn code() -> Option<ErrorCode> {
    // ...
  }
}
```

Reply:
Anyway, the rationale is that, at this low level you get the response as long as the request succeeded. The response has the contents of the response.

Reply:
The problem is that this is not "low level" - this standard Core NATS API for request/reply pattern. 
And I stand by the opinion that `no responders` is an Error.

Reply:
> The problem is that this is not "low level" - this standard Core NATS API for request/reply pattern.
And I stand by the opinion that no responders is an Error.

If we leave it as is, with special casing no responders everything except no responders will be catchable via `Message::status`, `jetstream::Response::Error::status` and `jetstream::Error::status`.